app/blister_api/actions.py

```diff
@@ -0,0 +1,251 @@
+from flask import g
+from flask_restful import abort
+
+from app import db
+from models import BucketListItem, User, BucketList
+
+
+# Database save, delete and update functionality.
+
+
+def save(record):
+    db.session.add(record)
+    db.session.commit()
+
+
+def delete(record):
+    db.session.delete(record)
+    db.session.commit()
+
+
+def update():
+    db.session.commit()
+
+# Ends here.
+# -------------------------------------------------
+# Actions pertaining to users
+
+
+def register_user(data):
+    '''
+    Parameters:
+    Data here is the JSON Object representing
+    the request sent with the URI.
+
+    Requires:
+    username and password to perform user registeration.
+
+    Returns:
+    The actual user with a 201 Success response for
+    the POST.
+    '''
+    username = data.get('username')
+    password = data.get('password')
+    if not username or not password:
+        abort(400, message='Please enter a username and password to register.')
+    username_exists = User.query.filter_by(username=username).first()
+    if username_exists:
+        abort(400, message='Username added already exist.')
+    user = User(username=username, password=password)
+    save(user)
+    return user
+
+
+def login_user(data):
+    """
+    Parameters:
+    Data here is the JSON Object representing
+    the request sent with the URI with actual credentials
+    in the database.
+
+    Requires:
+    username and password to perform login.
+
+    Returns:
+    The user.
+    If the return is a message, it returns the error message.
+
+    """
+    username = data.get('username')
+    password = data.get('password')
+    if not username or not password:
+        abort(400, message='Please enter a username and password to log in.')
+    user = User.query_user(username, password)
+    if type(user) == str:
+        abort(401, message=user)
+    else:
+        return user
+
+# Ends here.
+# -------------------------------------------------
+# C.R.U.D Operations for the API Begin here
+# =================================================
+# CREATE bucket list and bucket list items
+
+
+def create_bucketlist(data):
+    """
+    Attributes:
+    Title: Title of the bucket list
+    Description : Description of the bucket list.
+    """
+    if not data:
+        abort(400, message='Please add information for your bucketlist.')
+    if not data.get('title'):
+        abort(400, message='Please provide a title for your bucketlist')
+
+    title = data.get('title')
```

Comment:
Get title before line 94, then you can use `title` for the `if` check to avoid calling the get statement twice.